doc/code/qml_templates.rst

```diff
@@ -42,3 +42,10 @@ Custom templates can be constructed using the template decorator.
     :toctree:
 
     pennylane.templates.template
+
+Utility functions for input checks
+----------------------------------
+
+.. automodapi:: pennylane.templates.utils
+    :no-heading:
+    :include-all-objects:
```

Comment:
Weirdly, this doesn't seem to render for me when I build it locally 🤔 I get the module docstring, but none of the functions appear.

![image](https://user-images.githubusercontent.com/2959003/78019315-033ff300-7397-11ea-9ce5-70c4a53dcc9d.png)

Reply:
Oh, this is because the functions in `qml.templates.utils` are private functions! Unfortunately, it doesn't look like `automodapi` supports private functions. We can get around this by building the toctree manually.

Here, we can replace `automodapi` with `automodule`:

```rest
.. automodule:: pennylane.templates.utils
```

And then, in the `pennylane/templates/utils.py`, in the module docstring,
```rest
Utility functions provided for the templates. These are useful for standard input checks,
for example to make sure that arguments have the right shape, range or type.

.. currentmodule:: pennylane.templates.utils
.. autosummary::
    :toctree: api

    _check_no_variable
    _check_wires
    _get_shape
    _check_shape
    _check_shapes
    _check_is_in_options
    _check_type
    _check_number_of_layers
```

and it will look like this:
![image](https://user-images.githubusercontent.com/2959003/78019978-1f905f80-7398-11ea-9831-4631d9d72f82.png)

Reply:
After seeing your other comment, the other solution is to make them non-private 🙂

Reply:
Yes, I think that is cleaner, since we are making them public through the docs...